internal/k8s/appprotectdos/app_protect_dos_resources.go

```diff
@@ -0,0 +1,151 @@
+package appprotectdos
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var appProtectDosPolicyRequiredFields = [][]string{
+	{"spec", "mitigation_mode"},
+	{"spec", "automation_tools_detection"},
+	{"spec", "tls_fingerprint"},
+	{"spec", "signatures"},
+	{"spec", "bad_actors"},
+}
+
+var appProtectDosLogConfRequiredFields = [][]string{
+	{"spec", "content"},
+	{"spec", "filter"},
+}
+
+func validateRequiredFields(policy *unstructured.Unstructured, fieldsList [][]string) error {
+	for _, fields := range fieldsList {
+		field, found, err := unstructured.NestedMap(policy.Object, fields...)
+		if err != nil {
+			return fmt.Errorf("Error checking for required field %v: %w", field, err)
+		}
+		if !found {
+			return fmt.Errorf("Required field %v not found", field)
+		}
+	}
+	return nil
+}
+
+func validateRequiredFieldsNoCopy(policy *unstructured.Unstructured, fieldsList [][]string) error {
+	for _, fields := range fieldsList {
+		field, found, err := unstructured.NestedFieldNoCopy(policy.Object, fields...)
+		if err != nil {
+			return fmt.Errorf("Error checking for required field %v: %w", field, err)
+		}
+		if !found {
+			return fmt.Errorf("Required field %v not found", field)
+		}
+	}
+	return nil
+}
+
+// ValidateAppDosProtectLogConf validates LogConfiguration resource
+func ValidateAppProtectDosLogConf(logConf *unstructured.Unstructured) error {
+	lcName := logConf.GetName()
+	err := validateRequiredFields(logConf, appProtectDosLogConfRequiredFields)
+	if err != nil {
+		return fmt.Errorf("Error validating App Protect Dos Log Configuration %v: %w", lcName, err)
+	}
+
+	return nil
+}
+
+var (
+	logDstEx     = regexp.MustCompile(`(?:syslog:server=((?:\d{1,3}\.){3}\d{1,3}|localhost):\d{1,5})|stderr|(?:\/[\S]+)+`)
+	logDstFileEx = regexp.MustCompile(`(?:\/[\S]+)+`)
+)
+
+// ValidateAppProtectDosLogDestination validates destination for log configuration
+func ValidateAppProtectDosLogDestination(dstAntn string) error {
+	errormsg := "Error parsing App Protect Dos Log config: Destination must follow format: syslog:server=<ip-address | localhost>:<port> or stderr or absolute path to file"
```

Comment:
> or absolute path to file

what's the use case for saving log to file in a container?

Reply:
Those are all the option that can be in the directive of log destination.

I just copy that from nap:
// ValidateAppProtectLogDestination validates destination for log configuration
func ValidateAppProtectLogDestination(dstAntn string) error {
	errormsg := "Error parsing App Protect Log config: Destination must follow format: syslog:server=<ip-address | localhost>:<port> or stderr or absolute path to file"

So I thought to make the same error msg.

Reply:
I don't think it makes sense to save logs inside the container, but I'm not familiar with NAP/DOS use cases

Reply:
Its optional, so its give more flexibility for the log dest options

Reply:
Will remove this option

Reply:
Done